src/increments.jl

```diff
@@ -0,0 +1,156 @@
+# This file is a part of JuliaFEM.
+# License is MIT: see https://github.com/JuliaFEM/Materials.jl/blob/master/LICENSE
+
+module Increments
+
+import LinearAlgebra: norm
+import Tensors: tovoigt, fromvoigt
+
+import ..AbstractMaterial, ..integrate_material!
+import ..Utilities: Symm2
+
+export optimize_dstrain!, uniaxial_increment!, biaxial_increment!, stress_driven_uniaxial_increment!
+
+# The skeleton of the optimizer is always the same, so we provide it as a
+# higher-order function. The individual specific optimizer functions
+# (`update_dstrain!)` only need to define the "meat" of how to update `dstrain`.
+"""
+    optimize_dstrain!(material::AbstractMaterial, dstrain::AbstractVector{<:Real},
```

Comment:
Again touching the terminology, in my mind, we are solving a root-finding problem, instead of optimization. Would `find_dstrain!` be more descriptive?

Reply:
These methods are made to be able to easily simulate stress states produced by some of the most common test machines. For example in the push-pull machine (with smooth specimen) we know that the stress state is uniaxial (in the measuring area), so basically we have to find a strain increment that produces zero stress in the other directions.

Reply:
More descriptive, and shorter, too, so yes, let's change that.

(But as you say, it's only Newton-Raphson in the *currently implemented* strain/stress optimizers, so the wording should be kept ambiguous enough to allow for other kinds of optimization in the future. `find` is fine for that.)

Thanks for the description, that'll complement the docstrings nicely. :)